core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -64,11 +63,11 @@
 
 
     public TranslogWriter(ShardId shardId, long generation, FileChannel channel, Path path, ByteSizeValue bufferSize) throws IOException {
-        super(generation, channel, path, channel.position());
-        this.shardId = shardId;
+        super(generation, channel, path, channel.position(), shardId);
         this.outputStream = new BufferedChannelOutputStream(java.nio.channels.Channels.newOutputStream(channel), bufferSize.bytesAsInt());
         this.lastSyncedOffset = channel.position();
         totalOffset = lastSyncedOffset;
+        logger.trace("writer created. path [{}]", path);
```

Comment:
I think the logging in this class is unneeded. if you debug it you can add it back